### PR TITLE
feat: blacklist items from Discover page

### LINKED ISSUE #380: Reject !
### Description

Similar to the Request button, I would love to have a button that says Reject. Once rejected, I would no longer see that title in Discovery, Trending etc. 

### Desired Behavior

See the description. 

### Additional Context

_No response_

### Code of Conduct

- [X] I agree to follow Overseerr's Code of Conduct

### LINKED ISSUE #490: Hide Items
### Description

A per user option to "hide" unrequested media would be pretty cool.
or an admin level option to globally hide certain media.

### Desired Behavior

i.e adding a button to hide/blacklist something from the discover page so that it is no longer shown. 

(I'm thinking specifically of shows/movies that don't already exist, and will never be wanted)

### Additional Context

_No response_

### Code of Conduct

- [X] I agree to follow Overseerr's Code of Conduct

### PR DESCRIPTION
**Pull request by JoaquinOlivero**
[Feature Request] Hide Items #490

#### Description
Allow users with special permissions to add media items to a blacklist to prevent them from being displayed. 
The blacklist can be managed on the "Settings" page where media can be removed from it. In addition, if the user has permission to manage and view blacklisted items, there is a button on the movie/series card to remove it from the blacklist.

#### Screenshot (if UI-related)
output

#### To-Dos

- [X] Successful build `yarn build`
- [x] Translation keys `yarn i18n:extract`
- [x] Database migration (if required)

#### Issues Fixed or Closed

- Fixes #490
- Fixes #380

### CODE REVIEW

**Comment by Fallenbagel** [comment]
Is this PR complete?

**Comment by JoaquinOlivero** [comment]
> Is this PR complete? @JoaquinOlivero

Yes, it's complete

**Comment by Fallenbagel** [comment]
> > Is this PR complete? @JoaquinOlivero
> 
> Yes, it's complete

Okay I'm switching it to ready for review and I'll review it sometime this weekend c:

File: src/components/BlacklistModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      enterTo="opacity-100"
+      leave="transition-opacity duration-300"
+      leaveFrom="opacity-100"
+      leaveTo="opacity-0"
+      show={true}
+    >
+      <Modal
+        backgroundClickable
+        title={`Blacklist ${title}`}
+        subTitle="Are you sure you want to proceed?"
```
Comment (Gauvino) [inline src/components/BlacklistModal/index.tsx]: Could you add the translation key please ?

**Review by Gauvino — APPROVED** [review]
Test it the only missing point is the translation keys but everything else works fine !

Edit : How can I remove a blacklisted media from blacklist ? @JoaquinOlivero

**Review by gauthier-th — APPROVED** [review]

**Review by gauthier-th — APPROVED** [review]

**Review by Fallenbagel — APPROVED** [review]
Lgtm. Haven't tested it though

**Comment by oOStroudyOo** [comment]
@JoaquinOlivero My blacklist doesn't seem to act in the way your preview does.

Blacklisted items still appear on the discover pages, unless you remove all blacklist permissions from a user.
But by doing that then they also can not blacklist any items.

Am I missing something?

image
image

**Comment by gauthier-th** [comment]
> @JoaquinOlivero My blacklist doesn't seem to act in the way your preview does.
> 
> Blacklisted items still appear on the discover pages, unless you remove all blacklist permissions from a user. But by doing that then they also can not blacklist any items.
> 
> Am I missing something?
> 
> image image

This is the intended behavior.

Users with blacklist permissions can see and hide items so that regular users can't see these items anymore in the app.

**Comment by oOStroudyOo** [comment]
Understood.

Even as an admin, I'd still quite like these hidden from discover pages.
Maybe in the future we can enhance this feature.

**Comment by gauthier-th** [comment]
@oOStroudyOo see #1022 for this feature request

**Pull request merged**